lib/net.js

```diff
@@ -959,11 +959,12 @@ function internalConnect(
     // node::TCPConnectWrap isn't instantiated and attached to the JS instance
     // of TCPConnectWrap above until connect() is called. So don't set the init
     // trigger id until now.
-    setInitTriggerId(self[async_id_symbol]);
-    if (addressType === 4)
-      err = self._handle.connect(req, address, port);
-    else
-      err = self._handle.connect6(req, address, port);
+    defaultTriggerAsyncIdScope(self[async_id_symbol], function() {
+      if (addressType === 4)
+        err = self._handle.connect(req, address, port);
+      else
+        err = self._handle.connect6(req, address, port);
+    });
```

Comment:
I would rewrite this as:

```js
err = defaultTriggerAsyncIdScope(self[async_id_symbol], { self, addressType, req, address, port }, connectByType)
```

where `connectByType` is defined in the top level as:


```js
function connectByType({ self, addressType, req, address, port }) {
  if (addressType === 4)
    return self._handle.connect(req, address, port);
  else
    return self._handle.connect6(req, address, port);
}
```